tests/unit/test_index.py

```diff
@@ -302,43 +285,116 @@ def test_get_best_candidate__best_yanked_but_not_all(self, caplog):
 
 class TestPackageFinder:
 
-    @pytest.mark.parametrize('allow_yanked', [False, True])
-    def test_create__allow_yanked(self, allow_yanked):
+    @pytest.mark.parametrize('allow_all_prereleases, prefer_binary', [
+        (False, False),
+        (False, True),
+        (True, False),
+        (True, True),
+    ])
+    def test_create__candidate_evaluator(
+        self, allow_all_prereleases, prefer_binary,
+    ):
         """
-        Test that allow_yanked is passed to CandidateEvaluator.
+        Test that the candidate_evaluator attribute is set correctly.
         """
-        search_scope = SearchScope([], [])
         selection_prefs = SelectionPreferences(
-            allow_yanked=allow_yanked,
+            allow_yanked=True,
+            allow_all_prereleases=allow_all_prereleases,
+            prefer_binary=prefer_binary,
         )
+        target_python = TargetPython(py_version_info=(3, 7, 3))
+        target_python._valid_tags = ['tag1', 'tag2']
         finder = PackageFinder.create(
-            search_scope=search_scope,
+            search_scope=SearchScope([], []),
             selection_prefs=selection_prefs,
-            session=object(),
+            session=PipSession(),
+            target_python=target_python,
         )
         evaluator = finder.candidate_evaluator
-        assert evaluator._allow_yanked == allow_yanked
+        assert evaluator.allow_all_prereleases == allow_all_prereleases
+        assert evaluator._prefer_binary == prefer_binary
+        assert evaluator._supported_tags == ['tag1', 'tag2']
 
     def test_create__target_python(self):
```

Comment:
A test that checks that a default `target_python` is propagated correctly to the link evaluator may give us better coverage than these two, and rely on fewer implementation details.